SixTrack/CMakeLists.txt

```diff
@@ -0,0 +1,1079 @@
+		add_custom_command(TARGET dafor_copy_${loop} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy ${loop}n.f ${loop}/fort.1)
+
+		#do the copy of dafor (not symlink so it will work on windows too)
+		add_custom_command(TARGET dafor_copy_binary_${loop} POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy dafor ${loop}/dafor)
+
+		#run dafor
+		add_custom_command(TARGET dafor_run_${loop} COMMAND ${CMAKE_COMMAND} -E chdir ${loop} ./dafor > ${loop}_dafor.log)
+
+		#copy the output we need
+		add_custom_command(TARGET dafor_move_${loop} COMMAND ${CMAKE_COMMAND} -E copy ${loop}/fort.2 ${CMAKE_CURRENT_BINARY_DIR}/${loop}.f)
+
+		#remove the link and other files
+		add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${loop}.f COMMAND ${CMAKE_COMMAND} -E remove_directory ${CMAKE_BINARY_DIR}/${loop} DEPENDS dafor_remove_${loop})
+	else()
+		add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${loop}.f COMMAND ${CMAKE_COMMAND} -E copy ${loop}n.f ${loop}.f DEPENDS run_astuce)
+	endif()
+endforeach()
+
+###################################################################################################
+#linking
+###################################################################################################
+
+if(CERNLIB)
+#	if(EXISTS "/cern/pro/bin/cernlib")
+		#running on lxplus and friends
+#		SET_TARGET_PROPERTIES(SixTrack PROPERTIES LINK_FLAGS "`cernlib -u graflib packlib_noshift kernlib_noshift` -lg2c -lpthread")
+#	else()
+		add_library(g2c STATIC IMPORTED)
+		add_library(graflib STATIC IMPORTED)
+		add_library(grafX11 STATIC IMPORTED)
+		add_library(kernlib_noshift STATIC IMPORTED)
+		add_library(packlib_noshift STATIC IMPORTED)
+		add_library(X11 STATIC IMPORTED)
+		add_library(Xau STATIC IMPORTED)
+		add_library(xcb STATIC IMPORTED)
+		add_library(Xdmcp STATIC IMPORTED)
+
+		set_target_properties(g2c PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libg2c.a)
+		set_target_properties(graflib PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libgraflib.a)
+		set_target_properties(grafX11 PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libgrafX11.a)
+		set_target_properties(kernlib_noshift PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libkernlib_noshift.a)
+		set_target_properties(packlib_noshift PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libpacklib_noshift.a)
+		set_target_properties(X11 PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libX11.a)
+		set_target_properties(Xau PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libXau.a)
+		set_target_properties(xcb PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libxcb.a)
+		set_target_properties(Xdmcp PROPERTIES IMPORTED_LOCATION ${CMAKE_SOURCE_DIR}/lib32/libXdmcp.a)
+
+		find_library(pthread STATIC pthread)
+		find_library(dl STATIC dl)
+		find_library(gcc_eh STATIC gcc_eh)
+
+		target_link_libraries(SixTrack graflib grafX11 packlib_noshift kernlib_noshift g2c X11 xcb Xau Xdmcp dl pthread gcc_eh)
+#	endif()
+endif(CERNLIB)
```

Comment:
Exactly for CERNlib, and NAGlib, I think this is (and still should!) be using pre-build binaries from who-knows-where. Personally, I consider CERNlib obsolete...

Because of this, CERNlib should probably also conflict with 64-bit.